rfcs/0013-deprecate-currency-factory.md

```diff
@@ -0,0 +1,345 @@
+
+#### Consequences
+
+* Minor runtime migration will be needed for deleting Currency Factory's 
+storage.
+
+### Use an Externally Maintained Asset ID Reservation System
+
+Neither option presented seems to handle the reserving of asset IDs. Therefore, 
+neither of these will fully replace currency factory and would more correctly 
+replace our own pallet-assets. A replacement for pallet-assets may be a good 
+idea but would require much more substantial changes.
+
+* **Parity's pallet-assets**
+  
+  Forking this pallet and maintaining extra features we need (hold) is an 
+  option.
+  
+  > It is not clear from RFC why substrate assets are good. IMHO it should be 
+  clarified - year of man work to make UI/QA/FE/users happy is what we lack. 
+  IMHO better to have fork of assets and collaborate with ecosystem to tune 
+  wallets. Do not write own wallets.
+  
+  #### Pros
+  
+  * Very nice UI compatibility with Polkadot Dashboard
+
+  * Maintained by Parity
+
+  * Simple Interface
+
+  * Can run two pallets at same time for two purposes (external and mintable) 
+  (Demonstrated by Moonbeam)
+
+  * Handles WASM contracts addresses as IDs
+
+  * Permissionless when needed
+
+  * Supported by Polkadot wallets
+  
+  * Supports asset metadata
+  
+  #### Cons
+  
+  * No asset ID reservation system, only prevents duplicates
+
+  * Hold functionality is not present (We depend on this in several cases)
+
+* **Moonbeam's pallet-asset-manager**
+
+Note: Not looking at using or forking this directly, but instead how they use 
+different instances of the same pallet for mintable and external assets.
+
+  #### Pros
+  
+  * Maintained by moonbeam
+  
+  #### Cons
+  
+  * Partially specialized
+
+  * No asset ID reservation system, only prevents duplicates
+
+#### Consequences
+
+* Any of these options will require a difficult runtime migration as we move 
+existing storage for pallet-assets/pallet-asset-registry/pallet-currency-factory 
+to an alternative.
+
+## Conclusion
+
+After analyzing possible solutions, the best course of action to deprecate 
+Currency Factory is to maintain a fork of Parity's pallet-assets, embrace the 
+design Moonbeam uses to maintain two instances of their pallet, and to remove 
+our asset ID reservation system (for release two).
+
+Additionally, for foreign asset IDs, we can move to the standard of using XCM 
+Multi Locations. We can move to using a simple nonce for local asset IDs. Most
+of the foreign asset system is already implemented through Assets Registry and
+will not need significant changes.
+
+However, we currently lack the means of creating local assets. The proposal here
+is to use Assets Registry to register both local and foreign assets. Assets
+Registry will act as a routing layer between our two instances of pallet-assets.
+
+This will leave us with two independently stored asset ranges (local and 
+foreign) without the overhead of Currency Factory. 
+
+### Technical
+
+1. Asset IDs & Balances
+
```

Comment:
Instead of the complexity of the enum, routing can be handled internally by the pallet by confirm if some storage items exist or not - at the cost of storage access